website/addons/owncloud/views.py

```diff
@@ -0,0 +1,173 @@
+"""Views for the node settings page."""
+# -*- coding: utf-8 -*-
+import httplib as http
+
+from flask import request
+from modularodm import Q
+from modularodm.storage.base import KeyExistsException
+
+from framework.auth.decorators import must_be_logged_in
+
+from website.addons.base import generic_views
+from owncloud import Client as OwnCloudClient
+from website.addons.owncloud.model import OwnCloudProvider
+from website.addons.owncloud.settings import DEFAULT_HOSTS
+from website.addons.owncloud.serializer import OwnCloudSerializer
+from website.oauth.models import ExternalAccount
+from website.project.decorators import (
+    must_have_addon)
+from website.util import rubeus, api_url_for
+
+from website.addons.owncloud.utils import ExternalAccountConverter
+
+SHORT_NAME = 'owncloud'
+FULL_NAME = 'OwnCloud'
+
+owncloud_account_list = generic_views.account_list(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_import_auth = generic_views.import_auth(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_deauthorize_node = generic_views.deauthorize_node(
+    SHORT_NAME
+)
+
+owncloud_get_config = generic_views.get_config(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+def owncloud_root_folder(node_settings, auth, **kwargs):
+    """Build HGrid JSON for root node. Note: include node URLs for client-side
+    URL creation for uploaded files.
+    """
+    node = node_settings.owner
+    root = rubeus.build_addon_root(
+        node_settings=node_settings,
+        name=node_settings.folder_name,
+        permissions=auth,
+        user=auth.user,
+        nodeUrl=node.url,
+        nodeApiUrl=node.api_url,
+    )
+    return [root]
+
+@must_be_logged_in
+def owncloud_user_config_get(auth, **kwargs):
```

Comment:
If you modified the `ownCloudUserSettings.to_json` to make a super call then inject `hosts` in the return, you could be using the generic method of getting a user's config (see `website/addons/views/base.py:get_addon_user_config`). This could simplify both routing and frontend code, but isn't necessarily a blocker.